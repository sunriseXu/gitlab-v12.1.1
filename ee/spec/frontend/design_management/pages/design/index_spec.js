import { shallowMount } from '@vue/test-utils';
import DesignIndex from 'ee/design_management/pages/design/index.vue';

describe('Design management design index page', () => {
  let vm;

  function createComponent(loading = false) {
    const $apollo = {
      queries: {
        design: {
          loading,
        },
      },
    };

    vm = shallowMount(DesignIndex, {
      propsData: { id: '1' },
      mocks: { $apollo },
    });
  }

  it('sets loading state', () => {
    createComponent(true);

    expect(vm.element).toMatchSnapshot();
  });

  it('renders design index', () => {
    createComponent();

    vm.setData({
      design: {
        filename: 'test.jpg',
        image: 'test.jpg',
        updatedAt: '01-01-2019',
        updatedBy: {
          name: 'test',
        },
      },
    });

    expect(vm.element).toMatchSnapshot();
  });
});
