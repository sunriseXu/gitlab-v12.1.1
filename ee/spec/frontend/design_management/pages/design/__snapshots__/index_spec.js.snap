// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management design index page renders design index 1`] = `
<div
  class="design-detail fixed-top w-100 position-bottom-0 d-flex flex-column"
>
  <toolbar-stub
    id="1"
    name="test.jpg"
    updatedby="[object Object]"
  />
   
  <designimage-stub
    image="test.jpg"
    name="test.jpg"
  />
</div>
`;

exports[`Design management design index page sets loading state 1`] = `
<div
  class="design-detail fixed-top w-100 position-bottom-0 d-flex flex-column"
>
  <toolbar-stub
    id="1"
    isloading="true"
    name=""
    updatedby="[object Object]"
  />
   
  <designimage-stub
    image=""
    isloading="true"
    name=""
  />
</div>
`;
